routing/missioncontrol.go

```diff
@@ -303,3 +343,51 @@ func (m *MissionControl) reportEdgeFailure(failedEdge edge) {
 	history := m.createHistoryIfNotExists(failedEdge.from)
 	history.channelLastFail[failedEdge.channel] = now
 }
+
+// GetHistorySnapshot takes a snapshot from the current mission control state
+// and actual probability estimates.
+func (m *MissionControl) GetHistorySnapshot() *MissionControlSnapshot {
+	m.Lock()
+	defer m.Unlock()
+
+	log.Debugf("Requesting history snapshot from mission control: "+
+		"node_count=%v", len(m.history))
+
+	nodes := make([]*MissionControlNodeSnapshot, 0, len(m.history))
+
+	for v, h := range m.history {
+		channelSnapshot := make([]*MissionControlChannelSnapshot,
```

Comment:
What exact format do you suggest?

Reply:
So currently the probabilities of a channel are not symmetrical. This is something that I'd like to change in follow up prs, but for this pr I'd like to stay close to the previous version of mission control.

Reply:
i just mean to define it as []MissionControlSnapshot since the slice will already be on the heap, and then the MissionControlSnapshots will occupy a contiguous block of memory rather than doing one allocation per entry

Reply:
Done